services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py

```diff
@@ -0,0 +1,119 @@
+
+            payload = json_build_object('table', TG_TABLE_NAME,
+                                        'action', TG_OP,
+                                        'data', row_to_json(record));
+
+            PERFORM pg_notify('{DB_CHANNEL_NAME}', payload::text);
+
+            RETURN NULL;
+        END;
+    $$ LANGUAGE plpgsql;
+    """
+
+    trigger_registration_query = f"""
+    CREATE TRIGGER {DB_TRIGGER_NAME}
+    AFTER UPDATE OF outputs ON comp_tasks
+        FOR EACH ROW
+        WHEN (OLD.outputs::jsonb IS DISTINCT FROM NEW.outputs::jsonb)
+        EXECUTE PROCEDURE {DB_PROCEDURE_NAME}();
+    """
+    async with db_engine.acquire() as conn:
+        await conn.execute(notification_fct_query)
+        await conn.execute(trigger_registration_query)
+        
+
+async def unregister(app: web.Application) -> None:
+    # unlisten_query = f"UNLISTEN {DB_CHANNEL_NAME}"
+    drop_trigger_query = f"""
+    DROP TRIGGER {DB_TRIGGER_NAME} on comp_tasks;
+    """
+
+    db_engine: Engine = app[APP_DB_ENGINE_KEY]
+    async with db_engine.acquire() as conn:
+        # await conn.execute(unlisten_query)
+        await conn.execute(drop_trigger_query)
+
+async def listen(app: web.Application):
+    listen_query = f"LISTEN {DB_CHANNEL_NAME}"
+    db_engine: Engine = app[APP_DB_ENGINE_KEY]
+    async with db_engine.acquire() as conn:
+        await conn.execute(listen_query)
+        msg = await conn.connection.notifies.get()
+        log.debug("DB comp_tasks.outputs Update: <- %s", msg.payload)
+        node = json.loads(msg.payload)
+        node_data = node["data"]
+        task_output = node_data["outputs"]
+        node_id = node_data["node_id"]
+        project_id = node_data["project_id"]
+        # find the user(s) linked to that project (why do I need to know the user?)
+        joint_table = user_to_projects.join(projects)
+        query = select([user_to_projects]).select_from(joint_table).where(projects.c.uuid == project_id)
+        async for row in conn.execute(query):
+            user_id = row["user_id"]
+            node_data = await projects_api.update_project_node_outputs(app, user_id, project_id, node_id, data=task_output)
+            #FIXME: this is not the final version
+            from .computation_subscribe import post_messages
+            messages = {"nodeUpdated": {"Node": node_id, "Data": node_data}}
+            await post_messages(app, user_id, messages)
+        
+
+async def comp_tasks_listening_task(app: web.Application) -> None:
+    log.info("starting comp_task db listening task...")
+    try:
+        await register_trigger_function(app)
+        log.info("listening to comp_task events...")
+        while True:
+            await listen(app)
+    except asyncio.CancelledError:
+        pass
+    finally:
+        await unregister(app)
+
+async def setup_comp_tasks_listening_task(app: web.Application):
+    app[APP_COMP_TASKS_LISTENING_KEY] = asyncio.get_event_loop().create_task(comp_tasks_listening_task(app))
```

Comment:
if you do not need to use ``app[APP_COMP_TASKS_LISTENING_KEY]``, why do you store it?
can just use ``task`` variable and use it within the context after the ``yield`` as well

Reply:
well I agree.. this comes from a copy/paste of mine where I was using it to check for the task state. I'll remove it.